libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m

```diff
@@ -472,6 +470,9 @@ - (NSString *)isLoginRedirectUrl:(NSURL *)url
         BOOL foundStartURL = (startUrlValue != nil);
         BOOL foundValidEcValue = ([ecValue isEqualToString:@"301"] || [ecValue isEqualToString:@"302"]);
         if (foundStartURL && foundValidEcValue) {
+            if ([startUrlValue containsString:kVFSessionPrefix]) {
+                startUrlValue = [startUrlValue stringByReplacingOccurrencesOfString:kVFSessionPrefix withString:@""];
```

Comment:
Basically we need to extract the `url` value from the embedded `url` parameter in this case.

Reply:
I wonder: is the right thing to do here to get rid of this prefix, or get rid of the frontDoor VF ping functionality?

Reply:
We need the VF ping functionality to set cookies on those domains. Unless we start removing and setting cookies on the hybrid view ourselves. More info offline.

Reply:
I was just curious if the frontDoor redirect to that particular URL that's being removed, accomplished the same thing.  Seems like it might.

Reply:
Can I request that we do this either directly or indirectly through [`parseFrontDoorReturnUrlString`](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2438/files#diff-bafe72208f1ff3afa7b2024a36312efeR449)? It’s basically the purpose of that method, albeit currently for another edge case.

Reply:
(Or basically start the amendment from that [special case check section](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2438/files#diff-bafe72208f1ff3afa7b2024a36312efeR420) in `frontDoorUrlWithReturnUrl:returnUrlIsEncoded:createAbsUrl:`)